### ISSUE TITLE
Missing Week Column and Unclear Variable Labels in incremental_outcome Output

### ISSUE THREAD

**Issue by adeelh1** [google/meridian#623]
I used the incremental_outcome function to obtain contribution data at a weekly level. However, the downloaded file does not include a "week" column, making it unclear which data corresponds to each time period. Additionally, the column names are generic (e.g., column1, column2, etc.), without clear labels indicating which column represents the baseline and which correspond to other channels.
The function I used to get the results,
incremental_result = mmm_analyse.incremental_outcome(
    use_posterior = True,
    new_data = None,
    non_media_baseline_values = [0.5]*5,
    scaling_factor0 = 0.0,
    scaling_factor1 = 1.0,
    selected_geos = None,
    selected_times = None,
    media_selected_times = None,
    aggregate_geos = True,
    aggregate_times = True,
    inverse_transform_outcome = True,
    use_kpi = True,
    by_reach = True,
    include_non_paid_channels = True,
    batch_size = 32
) 
print("Incremental Outcome:", incremental_result)

Could you clarify if there is a way to:
1. Include a "week" column in the output?
2. Retrieve meaningful variable names instead of generic column labels?
3. Obtain the baseline results too ?

Thanks !

**Comment by cpulavarthi** [comment]
Hello @adeelh1,

Thank you for contacting us!

Please note that the `incremental_outcome()` method outputs a tensor of dimensions `(n_chains, n_draws, n_geos,n_times, n_channels)` depending on whether `aggregate_geos` and `aggregate_times` arguments are set to `True`. If they are `True` then the outcome dimensions will be `(n_chains, n_draws, n_channels)` as in your code. You may check our API documentation of the [`incremental_outcome`](https://developers.google.com/meridian/reference/api/meridian/analysis/analyzer/Analyzer#incremental_outcome) method for detailed information about the default parameters and the output it returns.

If you want week-wise incremental outcome, you will need to set `aggregate_times=False` which will give you the incremental outcome in your input data’s time-granularity, i.e., it will be week-wise incremental outcome if the initial time-granularity is weekly, else you may need to manually aggregate the results.

You can use the `baseline_summary_metrics()` method to obtain the baseline results (Ref - API Documentation of [Baseline Summary Metrics](https://developers.google.com/meridian/reference/api/meridian/analysis/analyzer/Analyzer#baseline_summary_metrics)).

Following is a sample code implementation to get your desired results:

```
model_analysis = analyzer.Analyzer(mmm)
incremental_tf = model_analysis.incremental_outcome(
                                                non_media_baseline_values = [0.5],
                                                aggregate_geos = True,
                                                aggregate_times = False,
                                                use_kpi = True,
                                                batch_size = 32
                                                )

incremental_df = pd.DataFrame(tf.reduce_mean(incremental_tf, axis=[0,1]).numpy())
incremental_df.columns = list(data.media_channel.values) + list(data.organic_media_channel.values) + list(data.non_media_channel.values)
incremental_df["Date"] = list(data.time.values)
incremental_df.set_index('Date', inplace=True)

baseline_tf = model_analysis.baseline_summary_metrics(confidence_level=0.9, aggregate_times=False)
baseline_df = baseline_tf.sel(metric='mean', distribution='posterior').to_dataframe()

final_df = incremental_df.copy()
final_df["Baseline_Outcome"] = baseline_df["baseline_outcome"]
final_df.head()
```

Feel free to reach out for any further queries.

Thank you

Google Meridian Support Team
